src/kubeobjects/legacy.go

```diff
@@ -1,27 +1,27 @@
 package kubeobjects
 
 import (
-	"k8s.io/client-go/discovery"
-	"k8s.io/client-go/rest"
+	"context"
+
+	apiv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const crdName = "oneagentapms.dynatrace.com"
+
 // CheckIfOneAgentAPMExists checks if a OneAgentAPM object exists
-func CheckIfOneAgentAPMExists(cfg *rest.Config) (bool, error) {
-	client, err := discovery.NewDiscoveryClientForConfig(cfg)
-	if err != nil {
-		return false, err
-	}
-	_, resourceList, err := client.ServerGroupsAndResources()
-	if err != nil {
+func CheckIfOneAgentAPMExists(clt client.Reader) (bool, error) {
+	var crd apiv1.CustomResourceDefinition
+
+	err := clt.Get(context.TODO(), client.ObjectKey{Name: crdName}, &crd)
```

Comment:
Does this work the same way now? It was a discovery client that checks for existence of specific APIs, but uses a normal client reader now. Check could also be done the same way as `checkCRD`

Reply:
I now adapted the change and made it similar to `checkCRD`, with the difference that I use `apiv1.CustomResourceDefinitionList` instead of `apiv1.DynakubeList.` So I think this should work as intended.

Reply:
Why not reuisng that function to be able to check any crd? Just passing the crd we want to check as parameter we could save one function and just have one that can check all the CRDs?